### PR TITLE
Endpoint to upload timetable .ics file

### LINKED ISSUE #52: Endpoint to store uploaded timetable file
**Describe the task that needs to be done.**
Create API endpoint that stores uploaded .ics file

**Describe how a solution to your proposed task might look like (and any alternatives considered).**
API endpoint can take in a uploaded .ics file and stores that file in a temporary folder. 

**Notes**
Discussed in meeting on 14/03, is just for when a file is uploaded, not provided via URL

### PR DESCRIPTION
**Pull request by kimslor**
## Description
*Please include a summary of the change and which issue number is fixed (e.g. Fixes #1). Please also include relevant motivation and context.*

This PR fixes issue #52
Implemented an endpoint to allow the uploading of a timetable .ics file.
This endpoint extracts the courses from the timetable and populates the user's courses just like the upload timetable via URL endpoint.

## How has this been tested?
*Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.*

A unit test was created using a .ics in the test resources folder. 

## Screenshots
*Provide screenshots of what has been implemented. Leave blank if not applicable*

## Checklist
*(Leave blank if not applicable)*

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/test/java/com/team701/buddymatcher/controllers/timetable/TimetableControllerIntegrationTest.java [hunk 1]
```diff
@@ -8,16 +8,22 @@
...
+import java.io.IOException;
+import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.Collections;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.when;
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
```
Comment (r055a) [inline src/test/java/com/team701/buddymatcher/controllers/timetable/TimetableControllerIntegrationTest.java]: It is not a good idea to import more than is required from org.springframework.test.web.servlet.request.MockMvcRequestBuilders. There are also 4 existing unused imports.

File: src/main/java/com/team701/buddymatcher/controllers/timetable/TimetableController.java [hunk 2]
```diff
@@ -14,11 +14,10 @@
 import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
 import org.springframework.web.server.ResponseStatusException;
 
-import java.io.ByteArrayInputStream;
-import java.io.IOException;
-import java.io.InputStream;
+import java.io.*;
```
Comment (r055a) [inline src/main/java/com/team701/buddymatcher/controllers/timetable/TimetableController.java]: It is not a good idea to import more than is required from java.io, and there is an existing case of importing more than is required from org.springframework.web.bind.annotation and java.util that could be refarctored to only include the necessary imports

File: src/main/java/com/team701/buddymatcher/controllers/timetable/TimetableController.java [hunk 3]
```diff
@@ -114,5 +113,20 @@ public TimetableController(TimetableService timetableService, ModelMapper modelM
         }
     }
 
+    @Operation(summary = "Post method to upload a course .isc file for the current user")
+    @PostMapping(path="users/upload/file")
+    public ResponseEntity<Void> uploadTimetableFile(@Parameter(hidden = true) @SessionAttribute("UserId") Long userId, @RequestParam("file") MultipartFile file) throws IOException {
```
Comment (r055a) [inline src/main/java/com/team701/buddymatcher/controllers/timetable/TimetableController.java]: This line extends beyond the line limit

**Review by R055A — CHANGES_REQUESTED** [review]
LGTM! However, I have left some suggestions for consideration, and I think some more documentation or comments could be better

File: src/main/java/com/team701/buddymatcher/controllers/timetable/TimetableController.java (hunk 3 above)
Comment (kimslor) [reply src/main/java/com/team701/buddymatcher/controllers/timetable/TimetableController.java]: Done.

File: src/test/java/com/team701/buddymatcher/controllers/timetable/TimetableControllerIntegrationTest.java (hunk 1 above)
Comment (kimslor) [reply src/test/java/com/team701/buddymatcher/controllers/timetable/TimetableControllerIntegrationTest.java]: Appropriate changes have been made.

File: src/main/java/com/team701/buddymatcher/controllers/timetable/TimetableController.java (hunk 2 above)
Comment (kimslor) [reply src/main/java/com/team701/buddymatcher/controllers/timetable/TimetableController.java]: The suggested changes have been made, except for the annotations import as the imports required were 7+ so I think it's better to keep it as is.

File: src/main/java/com/team701/buddymatcher/controllers/timetable/TimetableController.java [hunk 4]
```diff
@@ -14,6 +14,7 @@
 import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
```
Comment (r055a) [inline src/main/java/com/team701/buddymatcher/controllers/timetable/TimetableController.java]: It seems that one case of more importing than necessary has been overlooked

**Review by R055A — CHANGES_REQUESTED** [review]
Much better! I have left one further comment

File: src/main/java/com/team701/buddymatcher/controllers/timetable/TimetableController.java (hunk 4 above)
Comment (kimslor) [reply src/main/java/com/team701/buddymatcher/controllers/timetable/TimetableController.java]: Changes have been made.

**Review by R055A — APPROVED** [review]

**Pull request merged**